packages/@o3r/schematics/src/utility/component.ts

```diff
@@ -0,0 +1,72 @@
+import { Tree } from '@angular-devkit/schematics';
+import * as ts from 'typescript';
+import { DecoratorWithArg, getPropertyFromDecoratorFirstArgument, isDecoratorWithArg } from './ast';
+
+/**
+ * Returns true if `node` is the decorator of an Angular component
+ *
+ * @param node
+ */
+export const isNgClassDecorator = (node: ts.Node): node is DecoratorWithArg =>
+  isDecoratorWithArg(node)
+  && node.expression.expression.escapedText.toString() === 'Component';
+
+/**
+ * Returns true if `node` is the decorator of an Otter component
+ *
+ * @param node
+ */
+export const isO3rClassDecorator = (node: ts.Node): node is DecoratorWithArg =>
+  isDecoratorWithArg(node)
+  && node.expression.expression.escapedText.toString() === 'O3rComponent';
+
+/**
+ * Returns true if `node` is an Otter component
+ *
+ * @param node
+ */
+export const isO3rClassComponent = (node: ts.Node): node is ts.ClassDeclaration => {
+  if (!ts.isClassDeclaration(node)) {
+    return false;
+  }
+  const decorators = ts.getDecorators(node) || [];
+  return decorators.some(isO3rClassDecorator)
+    && decorators.some(isNgClassDecorator);
+};
+
+/**
+ * Returns Otter component information
+ *
+ * @param tree
+ * @param componentPath
+ */
+export const getO3rComponentInfo = (tree: Tree, componentPath: string) => {
+  const sourceFile = ts.createSourceFile(
+    componentPath,
+    tree.readText(componentPath),
+    ts.ScriptTarget.ES2020,
+    true
+  );
+
+  const o3rComponentDeclaration = sourceFile.statements.find(isO3rClassComponent);
+
+  if (!o3rComponentDeclaration) {
+    return;
```

Comment:
Would be good to return a good informative message, something like: couldn't get component information: missing O3rComponent or Component decorator